docs/running/cwl.rst

```diff
@@ -118,6 +118,18 @@ samples inputs, it could look something like:
 
 .. literalinclude:: ../../src/toil/test/docs/scripts/tutorial_cwlexample.py
 
+Running CWL workflows with InplaceUpdateRequirement
+---------------------------------------------------
+
+Some CWL workflows use the ``InplaceUpdateRequirement`` feature, which requires
+that operations on files have visible side effects that Toil's file store
+cannot support. If you need to run a workflow like this, you can make sure that
+all of your worker nodes have a shared filesystem, and use the
+``--bypass-file-store`` option to ``toil-cwl-runner``. This will make it leave
+all CWL intermediate files on disk and share them between jobs using file
+paths, instead of storing them in the file store and downloading them when jobs
+need them.
```

Comment:
Wouldn't that be faster for all users with a shared filesystem?

Reply:
Maybe in the case where you're also using a file job store on the shared filesystem, and symlinking everything, because you don't need to take the time to make those symlinks. And in the case where you have a lot of files, because with the file store you're reconstructing the directory structure for all the jobs that want it, instead of sharing it. But I could see cases where using S3 would be faster than bypassing the filestore through a shared filesystem, if your cluster is in AWS and has a pretty slow shared filesystem for home directories or something.

In general I see bypassing the filestore as a big hack to support some weird workflows that need a weird feature, and not really what we want most people to be doing most of the time. It definitely opens up new ways to break things: we don't really have a way for jobs that *don't* have `InplaceUpdateRequirement` to *not* see them, when we do it this way, so jobs might end up seeing files they shouldn't have, and then not work right when run on another cluster where there isn't a shared filesystem and the file store actually has to be used.